src/link.js

```diff
@@ -168,11 +167,20 @@ export default class Link extends Plugin {
 			}
 		} );
 
-		// Close on `ESC` press.
-		escPressHandler( {
-			emitter: balloonPanelView,
-			activator: () => balloonPanelView.isVisible,
-			callback: () => this._hidePanel( true )
+		// Focus the form if balloon panel is open and tab key has been pressed.
+		editor.ui.keystrokes.set( 'tab', ( data, cancel ) => {
```

Comment:
Related to https://github.com/ckeditor/ckeditor5-editor-classic/pull/44#pullrequestreview-17197218.

I think that the balloon could have its own keystroke handler. Then it wouldn't have to check whether it's visible and focused or not – that would be automatic.

I've been also thinking about some option to disable keystroke handler based on some condition. That may be needed at some point, but it seems that not yet.